src/store/memory.ts

```diff
@@ -334,7 +334,7 @@ export class MemoryStore implements IStore {
      * @param {string} eventType The event type being queried
      * @return {?MatrixEvent} the user account_data event of given type, if any
      */
-    public getAccountData(eventType: EventType | string): MatrixEvent | undefined {
+    public getAccountData(eventType: EventType | string): MatrixEvent {
```

Comment:
it can return undefined though? The docs would need updating otherwise.

Reply:
```
node_modules/matrix-js-sdk/lib/store/memory.d.ts:168:5 - error TS2416: Property 'getAccountData' in type 'MemoryStore' is not assignable to the same property in base type 'IStore'.
  Type '(eventType: string) => MatrixEvent | undefined' is not assignable to type '(eventType: string) => MatrixEvent'.
    Type 'MatrixEvent | undefined' is not assignable to type 'MatrixEvent'.
      Type 'undefined' is not assignable to type 'MatrixEvent'.
```

Reply:
This sounds like an issue with the upper interface, not with the store.

Reply:
To expand on what @turt2live mentioned, it sounds like we need to update `IStore` to allow `undefined` as well,

https://github.com/matrix-org/matrix-js-sdk/blob/871149912102b7fe65dc49df35be033f7cea5e62/src/store/index.ts#L160-L164